src/coreclr/jit/llvmcodegen.cpp

```diff
@@ -2514,17 +2518,21 @@ llvm::CallBase* Llvm::emitCallOrInvoke(llvm::FunctionCallee callee, ArrayRef<Val
     {
         catchLlvmBlock = m_EHDispatchLlvmBlocks[getCurrentProtectedRegionIndex()];
 
-        // Protected region index that is set in the emit context refers to the "logical" enclosing
-        // protected region, i. e. the one before funclet creation. But we do not need to (in fact,
-        // cannot) emit an invoke targeting block inside a different LLVM function.
-        if (catchLlvmBlock->getParent() != getCurrentLlvmFunction())
-        {
-            catchLlvmBlock = nullptr;
-        }
-        // No need to invoke no-throw functions.
-        else if (llvm::isa<Function>(callee.getCallee()) && llvm::cast<Function>(callee.getCallee())->doesNotThrow())
+        if (catchLlvmBlock != nullptr)
```

Comment:
Hmm, the state where we have a protected region without a dispatch block is arguably worse than checking that the protected region was unreachable...

I think I've come to a solution that lets us deal with dead throw helper blocks robustly:
1) In `emitJumpToThrowHelper` mark them: `throwBlock->bbFlags |= BBF_MARKED;`.
2) In the loop over additional code descriptors, check for the mark.

The we don't need to involve EH here at all.

Reply:
I'm not sure I've removed everything that can be, but have added the flag and checked it before `generateBlock`